### PR TITLE
[JENKINS-32525] - Handle null @AncestorInPath when selecting Downstream Jobs.

### PR DESCRIPTION
**Pull request by Dohbedoh**
https://issues.jenkins-ci.org/browse/JENKINS-32525

### CODE REVIEW

**Comment by Dohbedoh** [comment]
@reviewbybees

File: core/src/main/java/hudson/tasks/BuildTrigger.java [hunk 1]
```diff
@@ -367,6 +367,8 @@ public boolean showEvenIfUnstableOption(@CheckForNull Class<? extends AbstractPr
          * Form validation method.
          */
         public FormValidation doCheck(@AncestorInPath AbstractProject project, @QueryParameter String value) {
+            // JENKINS-32525: Check that it behaves gracefully for an unknown context
+            if (project == null) return FormValidation.ok();
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/tasks/BuildTrigger.java]: I would return a warning (as well as for the permission check case below)

**Comment by oleg-nenashev** [comment]
:bee: with a minor comment

File: core/src/main/java/hudson/tasks/BuildTrigger.java (hunk 1 above)
Comment (jglick) [inline core/src/main/java/hudson/tasks/BuildTrigger.java]: You can use `ok(String)` since there is not necessarily anything wrong.

Comment (oleg-nenashev) [inline core/src/main/java/hudson/tasks/BuildTrigger.java]: agreed. The only purpose is to provide extra troubleshooting info

Comment (Dohbedoh) [inline core/src/main/java/hudson/tasks/BuildTrigger.java]: Would something like "Ancestor/Context Unknown: the project specified cannot be validated" be helpful?

**Comment by oleg-nenashev** [comment]
retriggering the build, but :bee:

**Comment by jglick** [comment]
:bee: (test failure looks like a flake)

**Pull request merged**